### ISSUE TITLE
Add support for RegEx for domains in hiding rules (to match uBO's support)

### ISSUE THREAD

**Issue by DandelionSprout** [AdguardTeam/AdguardBrowserExtension#2559]
### Issue Details

While asterisk wildcarding for domains in hiding rules have had a wide consensus to be limited to 1 per entry (tied to TLDs), I've run into a situation on Nitter and by extension InvidioUS and Mastodon where one TLD wildcard simply isn't enough, and that additional flexible domain values would be a gamechanger for those sites in AdGuard.

As the situation currently stands, User Rules consider e.g. `/^nitter\..*$/#?#.timeline-item:has-text(tweet)` and `nitter.*.*#?#.timeline-item:has-text(brony)` to be invalid domains, so I've had to look at https://github.com/zedeus/nitter/wiki/Instances#public for working instances, then try to view a tweet on each domain shown as working to see if they actually do work (e.g. if PFPs show up, tweet shows up, may later have to test if embeds show up when linked to on Discord), and then use every single confirmed domain in the hiding rule, leading to stuff like `nitter.*,nitter.unixfox.eu,nitter.mint.lgbt,nitter.esmailelbob.xyz,nitter.d420.de,nitter.x86-64-unknown-linux-gnu.zip,nitter.woodland.cafe,nitter.dafriser.be,nitter.catsarch.com,nitter.perennialte.ch,nitter.salastil.com,nitter.uni-sonia.com,n.opnxng.com,nitter.ktachibana.party#?#.timeline-item:has-text(brony)`, which may possibly have to be re-tested every 4 weeks or so, if not more often.

### Proposed solution

The technical details behind https://github.com/uBlockOrigin/uBlock-issues/issues/2905 (which in turn were added in uBO 1.47.x onwards, almost a year ago), though I don't know much about the underlying code details.

Long story short, to add support for RegEx in domains, in order to do stuff like `/^nitter\..*$/#?#.timeline-item:has-text(brony)`, and to synchronise syntax support with uBO.

### Alternative solution

Double or multiple asterisk wildcarding for domains in hiding rules, to do stuff like `nitter.*.*##(...)`. Would be difficult for me to implement in multi-adblocker lists of mine, but I've been through worse projects.

**Comment by krystian3w** [comment]
This is duplicate of https://github.com/AdguardTeam/tsurlfilter/issues/89 and a few more.

**Comment by alexx7311** [comment]
@DandelionSprout Is the matter sufficiently covered in https://github.com/AdguardTeam/tsurlfilter/issues/89?

**Comment by DandelionSprout** [comment]
I *think* so, at least?
